From a9490d40a234f95b07cb0b3839e4c0ca4d3d828f Mon Sep 17 00:00:00 2001
From: Fabio Berton <fabio.berton@ossystems.com.br>
Date: Thu, 28 Sep 2017 11:33:21 -0300
Subject: [PATCH] Use BUILD_CC to compile native parts
Organization: O.S. Systems Software LTDA.

Set CC = ${BUILD_CC} ${BUILD_CFLAGS} ${BUILD_LDFLAGS} to all compile
native parts.

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Fabio Berton <fabio.berton@ossystems.com.br>
---
 tools/cert_create/Makefile | 2 +-
 tools/doimage/Makefile     | 2 +-
 tools/fiptool/Makefile     | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/cert_create/Makefile b/tools/cert_create/Makefile
index b0994b8..188a4b3 100644
--- a/tools/cert_create/Makefile
+++ b/tools/cert_create/Makefile
@@ -65,7 +65,7 @@ INC_DIR := -I ./include -I ${PLAT_INCLUDE} -I ${OPENSSL_DIR}/include
 LIB_DIR := -L ${OPENSSL_DIR}/lib
 LIB := -lssl -lcrypto
 
-HOSTCC ?= gcc
+HOSTCC ?= ${BUILD_CC} ${BUILD_CFLAGS} ${BUILD_LDFLAGS}
 
 .PHONY: all clean realclean
 
diff --git a/tools/doimage/Makefile b/tools/doimage/Makefile
index bc74369..0adce04 100644
--- a/tools/doimage/Makefile
+++ b/tools/doimage/Makefile
@@ -25,7 +25,7 @@ CFLAGS += ${DOIMAGE_CC_FLAGS}
 # could get pulled in from firmware tree.
 INCLUDE_PATHS = -I.
 
-CC := gcc
+CC := ${BUILD_CC} ${BUILD_CFLAGS} ${BUILD_LDFLAGS}
 RM := rm -rf
 
 .PHONY: all clean
diff --git a/tools/fiptool/Makefile b/tools/fiptool/Makefile
index e0e3923..533ae2b 100644
--- a/tools/fiptool/Makefile
+++ b/tools/fiptool/Makefile
@@ -29,7 +29,7 @@ endif
 
 INCLUDE_PATHS := -I../../include/tools_share
 
-HOSTCC ?= gcc
+HOSTCC ?= ${BUILD_CC} ${BUILD_CFLAGS} ${BUILD_LDFLAGS}
 
 .PHONY: all clean distclean
 
-- 
2.14.1

